src/sources/logplex.rs

```diff
@@ -59,38 +112,17 @@ impl SourceConfig for LogplexConfig {
     }
 }
 
-fn decode_message(body: Bytes, header_map: HeaderMap) -> Result<Vec<Event>, ErrorMessage> {
-    // Deal with headers
-    let msg_count = match usize::from_str(get_header(&header_map, "Logplex-Msg-Count")?) {
-        Ok(v) => v,
-        Err(e) => return Err(header_error_message("Logplex-Msg-Count", &e.to_string())),
-    };
-    let frame_id = get_header(&header_map, "Logplex-Frame-Id")?;
-    let drain_token = get_header(&header_map, "Logplex-Drain-Token")?;
-
-    emit!(HerokuLogplexRequestReceived {
-        msg_count,
-        frame_id,
-        drain_token
-    });
-
-    // Deal with body
-    let events = body_to_events(body);
-
-    if events.len() != msg_count {
-        let error_msg = format!(
-            "Parsed event count does not match message count header: {} vs {}",
-            events.len(),
-            msg_count
-        );
-
-        if cfg!(test) {
-            panic!(error_msg);
-        }
-        return Err(header_error_message("Logplex-Msg-Count", &error_msg));
+impl Auth {
+    fn as_header(&self) -> String {
+        let mut headers = HeaderMap::new();
+        headers.typed_insert(Authorization::basic(&self.username, &self.password));
```

Comment:
Why can't you return the `Authorization` directly?

Reply:
What benefit will give us [Authorization](https://docs.rs/headers/0.3.2/headers/authorization/struct.Authorization.html) instead `String`? On request we need to compare `&str` with something.